server/index.py

```diff
@@ -13,10 +15,12 @@ def status():
     return "it's up!"
 
 
-@app.route('/test-model', methods=['GET'])
-def get_text():
-    """Use pretrained model to generate text by the requested parameters."""
+@app.route('/lstm_trump_tweets_model', methods=['GET'])
+def get_lstm_trump_text():
+    """Use the LSTM trump tweets model to generate text."""
     data = json.loads(request.data)
-    string_length = data["string_length"]
-    seed_text = data["seed_text"]
-    return "deferring this until we have pretrained models!!!"
+    sl = data["string_length"]
+    st = data["seed_text"]
+    model = pretrained_factory.create_pretrained_LSTM_trump_tweets_generator()
```

Comment:
What's the error that occurs when this is moved outside of the request?

Reply:
I finally figured it out. there's a race condition! I'm still not sure why it exists. but I can fix it by running the server without threads.

Reply:
Oh interesting. I wonder if we’re doing that for Rocket 2 - as far as I know we’ve never seen a race condition, but we might be running it single-threaded.

Reply:
the error was appearing in tensorflow: Cannot interpret feed_dict key as Tensor: Tensor Tensor("Placeholder:0", shape=(4421, 500), dtype=float32) is not an element of this graph.

Reply:
and appeared when I moved the pretrained generator call outside the endpoint, or if I made a request a second time... weird.

Reply:
ok, in case you're curious, I know why this happens. with every http request, flask makes a new thread which creates it's own tensor flow session - one which doesn't have the models loaded, so the call fails of course. This medium article explains it well: https://kobkrit.com/tensor-something-is-not-an-element-of-this-graph-error-in-keras-on-flask-web-server-4173a8fe15e1